app/modules/collaborations/models.py

```diff
@@ -17,13 +51,126 @@ class Collaboration(db.Model, HoustonModel):
     guid = db.Column(
         db.GUID, default=uuid.uuid4, primary_key=True
     )  # pylint: disable=invalid-name
-    title = db.Column(db.String(length=50), nullable=False)
+    title = db.Column(db.String(length=50), nullable=True)
+
+    collaboration_user_associations = db.relationship(
+        'CollaborationUserAssociations', back_populates='collaboration'
+    )
+
+    def __init__(
+        self, user_guids=None, approval_states=None, is_initiator=None, **kwargs
+    ):
+        # I think this can be adapted to allow >2 User collabs, opening the door for Org-level.
+        if user_guids is None or len(user_guids) != 2:
+            raise ValueError(
+                '__init__ Collaboration: Creating a collaboration requires a user_guids list.'
+            )
+
+        # You can create a collaboration where both users 'approved'
+        if approval_states is not None and len(approval_states) != len(user_guids):
+            raise ValueError(
+                '__init__ Collaboration: Length of approval_states was not equal to provided user list.'
+            )
+
+        if is_initiator is not None and len(is_initiator) != len(user_guids):
+            raise ValueError(
+                '__init__ Collaboration: Length of is_initiator states was not equal to provided user list.'
+            )
+
+        for i in range(len(user_guids)):
```

Comment:
would prefer i to have a better name user_guid?

Reply:
It's not actually the guid, just the index. Since I want to access the same index on multiple optional lists I've just got the number. I do call my index(s) i,j,k just out of habit and I could rename it 'index' or 'guid_index'. Happy to make a change if that would be more readable.

Reply:
Yeah, I find that there's always something more readable than i j k etc.